### PR TITLE
Keep the OBS data in cache/

### PR DESCRIPTION
**Pull request by mvidner**
The code is in https://github.com/mvidner/rpm-build-dependencies/commit/2cf5cf3bf23b8f37088a0ca2ea1d0c7c89e6bd1c, the rest is the cached data.

### CODE REVIEW

**Comment by mvidner** [comment]
@jreidinger the cache should be more useful and easier to understand this way

File: README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[bsc#999203](https://bugzilla.suse.com/show_bug.cgi?id=999203)
+
+```console
+$ ruby print_deps.rb YaST:Head openSUSE_Factory x86_64 yast_deps.yaml
+```
```
Comment (jreidinger) [inline README.md]: OK, you open pandora box. If you would like to add README, then it should at least contain, what is this project good for, its features and describe terms used there ( like definition of critical path ).

ftw for example I think it is more useful to show it for openSUSE:Factory, as when I plan to introduce it, I would like to show it for factory.

File: print_deps.rb [hunk 2]
```diff
@@ -15,8 +17,37 @@ def initialize(name, depends)
   end
 end
 
+def cache_dir
+  name = File.expand_path("../cache", __FILE__)
```
Comment (jreidinger) [inline print_deps.rb]: so if script in future will live in /usr/bin, then it create stuff there? I hope not.

File: print_deps.rb [hunk 3]
```diff
@@ -15,8 +17,37 @@ def initialize(name, depends)
...
+  FileUtils.mkdir_p name unless File.directory? name
+  name
```
Comment (jreidinger) [inline print_deps.rb]: what about caching this method? It needs to be called only once.

**Review by jreidinger — CHANGES_REQUESTED** [review]
I am really against having cache in git repository. It significantly increase git repo size and it is useless, as you want recent result, so you checkout new version and still are interested in new results.

File: print_deps.rb (hunk 2 above)
Comment (mvidner) [reply print_deps.rb]: Yes, we will fix that once we move the program. Not later, not sooner.

File: print_deps.rb (hunk 3 above)
Comment (mvidner) [reply print_deps.rb]: Uh, how do you memoize a global method? With a global variable?

File: README.md (hunk 1 above)
Comment (mvidner) [reply README.md]: Improved. Regarding openSUSE:Factory, let's change that once it is ready, like with the cache dir placement.

File: print_deps.rb (hunk 3 above)
Comment (jreidinger) [reply print_deps.rb]: yes, for example. this cache_dir call is used very often.

**Review by jreidinger — APPROVED** [review]

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**